File: backend/src/main/kotlin/com/example/backend/services/AuditLogService.kt
package com.example.backend.services

  import com.example.backend.models.AuditLog
  import com.example.backend.repositories.AuditLogRepository
  import org.springframework.stereotype.Service

  @Service
  class AuditLogService(
    private val auditLogRepository: AuditLogRepository
  ) {
    fun saveLog(userActionLog: AuditLog) {
      auditLogRepository.save(userActionLog)
    }
  }

File: backend_py/pharmaPython/app/services/audit_log_service.py
from sqlalchemy.orm import Session

from app.models.audit_log import AuditLog
from app.repositories.audit_log_repository import AuditLogRepository


class AuditLogService:
  def __init__(self, db: Session):
    self.db = db
    self.auditLogRepository = AuditLogRepository(db)

  def save_log(self, log: AuditLog):
    return self.auditLogRepository.save(self.db, log)
